src/handlers/dns_xact/DnsXactStreamHandler.h

```diff
@@ -0,0 +1,291 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#pragma once
+
+#include "AbstractMetricsManager.h"
+#include "PcapInputStream.h"
+#include "StreamHandler.h"
+#include "dnstap.pb.h"
+#include "querypairmgr.h"
+#include <Corrade/Utility/Debug.h>
+#include <bitset>
+#include <limits>
+#include <string>
+#include "visor_dns/DnsLayer.h"
+
+namespace visor::input::dnstap {
+class DnstapInputEventProxy;
+}
+
+namespace visor::handler::dnsxact {
+
+using namespace visor::dns;
+using namespace visor::input::pcap;
+using namespace visor::input::dnstap;
+
+enum QRXact {
+    query = 0,
+    response = 1
+};
+
+enum Protocol : uint64_t {
+    DNSTAP_UDP = dnstap::SocketProtocol::UDP,
+    DNSTAP_TCP = dnstap::SocketProtocol::TCP,
+    DNSTAP_DOT = dnstap::SocketProtocol::DOT,
+    DNSTAP_DOH = dnstap::SocketProtocol::DOH,
+    PCPP_TCP = pcpp::TCP,
+    PCPP_UDP = pcpp::UDP
+};
+
+class DnsXactMetricsBucket final : public visor::AbstractMetricsBucket
+{
+protected:
+    mutable std::shared_mutex _mutex;
+
+    Quantile<uint64_t> _dnsXactFromTimeUs;
+    Quantile<uint64_t> _dnsXactToTimeUs;
+    Quantile<double> _dnsXactRatio;
+    TopN<std::string> _dns_slowXactIn;
+    TopN<std::string> _dns_slowXactOut;
+
+    struct counters {
+        Counter xacts_total;
+        Counter xacts_in;
+        Counter xacts_out;
+        Counter xacts_timed_out;
+        Counter filtered;
+        counters()
+            : xacts_total("dns", {"xact", "counts", "total"}, "Total DNS transactions (query/reply pairs)")
+            , xacts_in("dns", {"xact", "in", "total"}, "Total ingress DNS transactions (host is server)")
+            , xacts_out("dns", {"xact", "out", "total"}, "Total egress DNS transactions (host is client)")
+            , xacts_timed_out("dns", {"xact", "counts", "timed_out"}, "Total number of DNS transactions that timed out")
+            , filtered("dns", {"wire_packets", "filtered"}, "Total DNS wire packets seen that did not match the configured filter(s) (if any)")
```

Comment:
For consistency, I'm thinking we should name this metric something like `dns_xact_packets_filtered` (or `dns_xact_wire_packets_filtered`). It's not ideal but at least it's consistent.

Reply:
done as dns_xact_packets_filtered